src/Servers/Kestrel/test/InMemory.FunctionalTests/ResponseTests.cs

```diff
@@ -3516,6 +3512,217 @@ public async Task ResponseBodyPipeCompleteWithoutExceptionWritesDoNotThrow()
+                        "Host:",
+                        "",
+                        "");
+                    await connection.Receive(
+                        "HTTP/1.1 200 OK",
+                        $"Date: {server.Context.DateHeaderValue}",
+                        "Transfer-Encoding: chunked",
+                        "",
+                        "1",
+                        "a",
+                        "0",
+                        "",
+                        "");
+
+                    await connection.Send(
+                        "GET / HTTP/1.1",
+                        "Host:",
+                        "",
+                        "");
+                    await connection.Receive(
+                        "HTTP/1.1 200 OK",
+                        $"Date: {server.Context.DateHeaderValue}",
+                        "Content-Length: 0",
+                        "",
+                        "");
+                }
+                await server.StopAsync();
+            }
+        }
+
+        [Fact]
+        public async Task ResponseStartCalledAndAutoChunkStateIsResetWithMultipleReqeusts()
+        {
+            using (var server = new TestServer(async httpContext =>
+            {
+                var memory = httpContext.Response.BodyWriter.GetMemory();
+                Encoding.ASCII.GetBytes("a").CopyTo(memory);
+                httpContext.Response.BodyWriter.Advance(1);
+                await httpContext.Response.BodyWriter.FlushAsync();
+
+            }, new TestServiceContext(LoggerFactory)))
+            {
+                using (var connection = server.CreateConnection())
+                {
+                    await connection.Send(
+                        "GET / HTTP/1.1",
+                        "Host:",
+                        "",
+                        "");
+                    await connection.Receive(
+                        "HTTP/1.1 200 OK",
+                        $"Date: {server.Context.DateHeaderValue}",
+                        "Transfer-Encoding: chunked",
+                        "",
+                        "1",
+                        "a",
+                        "0",
+                        "",
+                        "");
+
+                    await connection.Send(
+                        "GET / HTTP/1.1",
+                        "Host:",
+                        "",
+                        "");
+                    await connection.Receive(
+                        "HTTP/1.1 200 OK",
+                        $"Date: {server.Context.DateHeaderValue}",
+                        "Transfer-Encoding: chunked",
+                        "",
+                        "1",
+                        "a",
+                        "0",
+                        "",
+                        "");
+                }
+                await server.StopAsync();
+            }
+        }
+
+        [Fact]
+        public async Task ResponseStartCalledStateIsResetWithMultipleReqeusts()
```

Comment:
@halter73 is this sufficient for the type of test you wanted?

Reply:
Yep. I would like to to flip back to chunked one last time to ensure the content length response to chunked response transition also works.

You can shorten the test slightly by pipelining the requests like we do in ZeroContentLengthNotSetAutomaticallyForCertainStatusCodes.

Reply:
Its in a for loop so it already does the Chunked to Content Length transition.